infra/token-transfer-server/src/lib/transfer.js

```diff
@@ -18,47 +21,50 @@ const ConfirmationTimeoutSec = 20 * 60 * 60
 /**
  * Helper method to check the validity of a transfer request.
  * Throws an exception in case the request sis invalid.
- * @param userId
  * @param grantId
  * @param amount
- * @param {Transfer} transfer: Optional. The pending transfer to check for.
- * @returns {Promise<void>}
+ * @returns {Promise<Grant>}
  * @private
  */
-async function _checkTransferRequest(userId, grantId, amount, transfer = null) {
-  // Check the user exists.
-  const user = await User.findOne({
-    where: { id: userId }
-  })
-  if (!user) {
-    throw new Error(`No user found with id ${userId}`)
-  }
-
+async function _checkTransferRequest(grantId, amount) {
   // Load the grant and check there are enough tokens available to fullfill the transfer request.
   const grant = await Grant.findOne({
     where: {
       id: grantId
-      // FIXME(franck): add this condition after changing grant table to have userId
-      //userId
-    }
+    },
+    include: [{ model: Transfer }]
   })
   if (!grant) {
-    throw new Error(
-      `Could not find specified grant id ${grantId} for user ${user.email}`
-    )
+    throw new Error(`Could not find specified grant id ${grantId}`)
   }
 
-  // TODO(franck/tom): Replace with a call to the logic for calculating tokens available.
-  //   Note: If transfer arg is not null, make sure to not double count that amount.
-  const placeholderAvailable = () => {
-    return grant.amount
-  }
-  const available = placeholderAvailable(userId, grantId, transfer)
+  // Sum the amount from transfers that are in a pending or complete state
+  const pendingOrCompleteTransfers = [
+    enums.TransferStatuses.Enqueued,
+    enums.TransferStatuses.Paused,
+    enums.TransferStatuses.WaitingConfirmation,
+    enums.TransferStatuses.Success
+  ]
+
+  const vested = vestedAmount(grant.get({ plain: true }))
+  logger.info('Vested tokens', vested)
+
+  const pendingOrCompleteAmount = grant.Transfers.reduce((total, transfer) => {
+    if (pendingOrCompleteTransfers.includes(transfer.status)) {
+      return (total += BigNumber(transfer.amount))
+    }
+    return total
+  }, 0)
+  logger.info('Pending or transferred tokens', pendingOrCompleteAmount)
+
+  const available = vested - pendingOrCompleteAmount
```

Comment:
Should this use BigNumber.minus method rather than JS '-' operator ?
Same comment for the line below. Should use BigNumber.gt